site/src/components/link.tsx

```diff
@@ -34,6 +34,7 @@ export const BaseLinkWithAnchor = forwardRef<
       scroll={scroll}
       shallow={shallow}
       locale={locale}
+      data-testid={other["data-testid"]}
```

Comment:
`[data-testid]` was previously not getting passed to the `a` tag rendered by `Link`